iocore/eventsystem/Thread.cc

```diff
@@ -57,6 +57,13 @@ Thread::Thread()
 Thread::~Thread()
 {
   ink_release_assert(mutex->thread_holding == (EThread *)this);
+
+  if (ink_thread_getspecific(Thread::thread_data_key) == this) {
```

Comment:
Is this to avoid race conditions on shutdown?

Reply:
Yes